pkg/activator/net/revision_backends.go

```diff
@@ -119,12 +107,94 @@ type destHealth struct {
 	health bool
 }
 
+func (rw *revisionWatcher) hasHealthyClusterIP() bool {
+	return rw.healthyClusterDest != ""
+}
+
+func (rw *revisionWatcher) revisionReady() bool {
+	return len(rw.dests) > 0
+}
+
+func (rw *revisionWatcher) getK8sPrivateService() (*corev1.Service, error) {
+	selector, err := labels.Parse(fmt.Sprintf("%s=%s,%s=%s",
```

Comment:
can we use
```
metav1.LabelSelector{
	MatchLabels: map[string]string{
		serving.RevisionLabelKey: ...,
	}
}
```
to avoid parsing and `err` handling.